### PR TITLE
SONARPY-1796 Infer types for set, dict and tuple literals

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**
Tuple content type is inferred immediately if (and unique) available as tuple are immutables.
This should be seen as a baby step/low-hanging fruit. I'd expect a dedicated effort to tackle any more complex case.

Also, this PR does not deal with display name/representation of those generic types.

Not 100% related to the ticket, but I also took the opportunity to add some missing test for some already covered literal types (str, numeric literals and None).

### CODE REVIEW

File: python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java [hunk 1]
```diff
@@ -85,6 +91,35 @@ public void visitStringLiteral(StringLiteral stringLiteral) {
...
+  public void visitTuple(Tuple tuple) {
+    super.visitTuple(tuple);
+    List<PythonType> contentTypes = tuple.elements().stream().map(Expression::typeV2).distinct().toList();
+    List<PythonType> attributes = List.of();
+    if (contentTypes.size() == 1 && !contentTypes.get(0).equals(PythonType.UNKNOWN)) {
+      attributes = contentTypes;
+    }
+    ModuleType builtins = this.projectLevelTypeTable.getModule(BUILTINS);
+    PythonType tupleType = builtins.resolveMember("tuple").orElse(PythonType.UNKNOWN);
+    ((TupleImpl) tuple).typeV2(new ObjectType(tupleType,  attributes, List.of()));
```
Comment (maksim-grebeniuk-sonarsource) [inline python-frontend/src/main/java/org/sonar/python/semantic/v2/TypeInferenceV2.java:104]: using `List.of()` could be dangerous since it is immutable. I'd prefer to use array lists instead

**Review by maksim-grebeniuk-sonarsource — APPROVED** [review]
LGTM
